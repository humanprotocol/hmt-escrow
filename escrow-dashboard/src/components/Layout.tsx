import * as React from 'react';
import { Box } from '@mui/material';
import Footer from './Footer';
import Header from './Header';

interface ILayout {
  children: React.ReactNode;
}

const Layout: React.FC<ILayout> = ({ children }): React.ReactElement => (
  <Box
    sx={{
      marginTop: '88px',
    }}
  >
    <Header />
    {children}
    <Footer />
  </Box>
);

export default Layout;
