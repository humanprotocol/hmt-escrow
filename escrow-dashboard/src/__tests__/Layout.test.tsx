import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import Layout from 'src/components/Layout';

describe('when rendered Layout component', () => {
  beforeAll(async () => {
    global.fetch = jest.fn().mockImplementationOnce(() =>
      Promise.resolve({
        status: 200,
        json: () =>
          Promise.resolve({
            market_data: {
              circulating_supply: 0,
              total_supply: 0,
              current_price: { usd: 0 },
              price_change_percentage_24h: 0,
            },
          }),
      })
    );
  });

  it('should render passed `children` prop', () => {
    act(() => {
      const { getByTestId } = render(
        <Layout>
          <div data-testid="root">
            <div data-testid="parent">
              <div data-testid="child">content</div>
            </div>
          </div>
        </Layout>
      );

      const root = getByTestId('root');
      const parent = getByTestId('parent');
      const child = getByTestId('child');
      expect(root).toContainElement(parent);
      expect(parent).toContainElement(child);
      expect(child).not.toContainElement(parent); // Pass
    });
  });
});

it('Layout component renders correctly, corresponds to the snapshot', () => {
  const tree = renderer
    .create(
      <Layout>
        <div data-testid="root">
          <div data-testid="parent">
            <div data-testid="child">content</div>
          </div>
        </div>
      </Layout>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
