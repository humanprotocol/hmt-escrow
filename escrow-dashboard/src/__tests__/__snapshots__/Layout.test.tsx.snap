// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiBox-root css-14uw52k"
>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-164jy24-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <div
          className="MuiBox-root css-1qtwkyj"
        >
          <div
            className="MuiBox-root css-70qvj9"
          >
            <img
              alt="logo"
              src="test-file-stub"
            />
            <h6
              className="MuiTypography-root MuiTypography-h6 css-i6uuzy-MuiTypography-root"
            >
              Scan
            </h6>
          </div>
          <div
            className="MuiBox-root css-84nudp"
          >
            <a
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-q1xa4c-MuiTypography-root-MuiLink-root"
              href="https://humanprotocol.org"
              onBlur={[Function]}
              onFocus={[Function]}
              target="_blank"
            >
              <p
                className="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                HUMAN Website
              </p>
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    data-testid="root"
  >
    <div
      data-testid="parent"
    >
      <div
        data-testid="child"
      >
        content
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-101hd6g"
  >
    <div
      className="MuiBox-root css-70qvj9"
    >
      <img
        alt="logo"
        src="test-file-stub"
      />
      <span
        className="MuiTypography-root MuiTypography-caption css-1ayj9iy-MuiTypography-root"
      >
        Terms and conditions
      </span>
    </div>
    <span
      className="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
    >
      © 
      2022
       HPF. HUMAN Protocol® is a registered trademark
    </span>
    <div
      className="css-19qjufr-MuiStack-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        fill="none"
        focusable="false"
        height="25"
        viewBox="0 0 27 25"
        width="27"
      >
        <path
          clipRule="evenodd"
          d="M13.1133 0C5.87195 0 0 5.63566 0 12.5889C0 18.1496 3.75712 22.8672 8.96894 24.5325C9.62504 24.6476 9.86414 24.2589 9.86414 23.9251C9.86414 23.6268 9.85287 22.8348 9.84643 21.7846C6.1988 22.545 5.42918 20.0969 5.42918 20.0969C4.83265 18.6426 3.97287 18.2562 3.97287 18.2562C2.78221 17.475 4.06303 17.4904 4.06303 17.4904C5.37927 17.5801 6.0716 18.7879 6.0716 18.7879C7.24132 20.7112 9.14041 20.1556 9.88829 19.8342C10.0074 19.0205 10.3456 18.4656 10.7207 18.1511C7.80888 17.8335 4.74731 16.7532 4.74731 11.9297C4.74731 10.5558 5.25851 9.43218 6.09736 8.55202C5.96212 8.23365 5.5121 6.95397 6.22536 5.2207C6.22536 5.2207 7.32666 4.88223 9.83194 6.51119C10.8777 6.23145 11.9999 6.09236 13.1149 6.08695C14.2283 6.09236 15.3505 6.23145 16.3978 6.51119C18.9015 4.88223 20.0004 5.2207 20.0004 5.2207C20.7161 6.95397 20.266 8.23365 20.1308 8.55202C20.9713 9.43218 21.4792 10.5558 21.4792 11.9297C21.4792 16.7656 18.4128 17.8297 15.4914 18.1411C15.9623 18.5298 16.3817 19.2979 16.3817 20.4725C16.3817 22.1547 16.3656 23.5125 16.3656 23.9251C16.3656 24.262 16.6015 24.6538 17.2673 24.5309C22.4735 22.8626 26.2282 18.148 26.2282 12.5889C26.2282 5.63566 20.3562 0 13.1133 0"
          fill="#858EC6"
          fillRule="evenodd"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        fill="none"
        focusable="false"
        height="20"
        viewBox="0 0 25 20"
        width="25"
      >
        <path
          d="M21.1774 1.675C19.5591 0.8925 17.8287 0.323815 16.0197 0C15.7975 0.415119 15.538 0.973462 15.359 1.41762C13.4359 1.11874 11.5306 1.11874 9.64287 1.41762C9.46395 0.973462 9.1985 0.415119 8.97434 0C7.16334 0.323815 5.431 0.894589 3.8127 1.67914C0.548586 6.77669 -0.336262 11.7476 0.106163 16.648C2.2711 18.3188 4.36918 19.3338 6.43185 19.9979C6.94113 19.2736 7.39535 18.5035 7.78664 17.692C7.04141 17.3994 6.32763 17.0382 5.6532 16.6189C5.83213 16.482 6.00714 16.3387 6.17623 16.1914C10.2898 18.1798 14.7593 18.1798 18.8237 16.1914C18.9948 16.3387 19.1697 16.482 19.3467 16.6189C18.6703 17.0403 17.9546 17.4014 17.2093 17.6941C17.6006 18.5035 18.0529 19.2757 18.5641 20C20.6288 19.3358 22.7288 18.3209 24.8937 16.648C25.4129 10.9672 24.0069 6.04194 21.1774 1.675ZM8.34706 13.6343C7.11221 13.6343 6.09953 12.4429 6.09953 10.9921C6.09953 9.54132 7.09059 8.34788 8.34706 8.34788C9.60357 8.34788 10.6162 9.53923 10.5946 10.9921C10.5965 12.4429 9.60357 13.6343 8.34706 13.6343ZM16.6529 13.6343C15.418 13.6343 14.4053 12.4429 14.4053 10.9921C14.4053 9.54132 15.3963 8.34788 16.6529 8.34788C17.9093 8.34788 18.922 9.53923 18.9004 10.9921C18.9004 12.4429 17.9093 13.6343 16.6529 13.6343Z"
          fill="#858EC6"
          opacity="0.8"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        fill="none"
        focusable="false"
        height="20"
        viewBox="0 0 26 20"
        width="26"
      >
        <path
          clipRule="evenodd"
          d="M25.9017 2.39701C25.0762 2.7466 24.1666 2.95803 23.2757 3.09772C23.6991 3.03131 24.3227 2.31611 24.572 2.02682C24.952 1.58563 25.2683 1.08262 25.4465 0.539924C25.4694 0.49718 25.4865 0.443749 25.4408 0.411691C25.3844 0.390319 25.3476 0.401768 25.3018 0.422377C24.3063 0.920807 23.2782 1.28184 22.183 1.53755C22.0915 1.55892 22.0236 1.53755 21.9672 1.48412C21.8765 1.38871 21.7915 1.29864 21.6943 1.22383C21.2349 0.858217 20.7405 0.571219 20.1831 0.359024C19.4524 0.080422 18.6531 -0.037125 17.8652 0.0101991C17.0994 0.05676 16.3467 0.259032 15.6667 0.590301C14.9851 0.923097 14.377 1.38565 13.8932 1.94362C13.3922 2.52067 13.0228 3.20229 12.8331 3.92513C12.6509 4.62277 12.6656 5.31279 12.7776 6.01731C12.7939 6.13409 12.7833 6.15012 12.6697 6.13409C8.38541 5.53873 4.83344 4.10679 1.95002 1.03835C1.82498 0.905541 1.75714 0.905541 1.65498 1.04904C0.393073 2.8153 1.00523 5.65246 2.58098 7.04699C2.79102 7.23323 3.0076 7.41871 3.23972 7.58816C3.15472 7.60496 2.10613 7.4981 1.17196 7.04699C1.04609 6.97295 0.983161 7.0157 0.972536 7.14775C0.961094 7.33857 0.97744 7.51413 1.00686 7.72174C1.24796 9.51166 2.57525 11.168 4.39374 11.8138C4.61032 11.8992 4.84897 11.9733 5.08108 12.0107C4.66672 12.0946 4.2409 12.1557 3.05174 12.0687C2.90462 12.0427 2.84741 12.1114 2.90462 12.2442C3.79793 14.5166 5.7243 15.1921 7.16764 15.5775C7.36052 15.6088 7.55341 15.6088 7.74792 15.6516C7.73566 15.6684 7.72422 15.6684 7.7136 15.6836C7.23466 16.3653 5.57473 16.8721 4.80239 17.1247C3.39664 17.5835 1.86584 17.7896 0.380813 17.6529C0.14298 17.6201 0.093125 17.6232 0.0285586 17.6529C-0.0368251 17.6903 0.0203856 17.7438 0.0963942 17.801C0.397977 17.9873 0.704463 18.1521 1.01749 18.3109C1.96146 18.778 2.94467 19.1498 3.97283 19.4154C9.28362 20.7847 15.267 19.7787 19.2555 16.0783C22.3873 13.1724 23.4841 9.16513 23.4841 5.1525C23.4841 4.99526 23.6836 4.90825 23.8013 4.82505C24.6161 4.25334 25.2634 3.56943 25.8707 2.80538C26.0071 2.63363 25.9998 2.48098 25.9998 2.41839C25.9998 2.4077 25.9998 2.39701 25.9998 2.39701C25.9998 2.3329 26.0071 2.35274 25.9017 2.39701Z"
          fill="#858EC6"
          fillRule="evenodd"
          opacity="0.8"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        fill="none"
        focusable="false"
        height="20"
        viewBox="0 0 26 20"
        width="26"
      >
        <path
          d="M1.72792 8.36813C8.47512 5.62492 12.9743 3.81642 15.2255 2.94265C21.6531 0.447838 22.9887 0.0144629 23.8592 0.000152579C24.0506 -0.00299481 24.4787 0.0412848 24.756 0.251262C24.9902 0.428562 25.0546 0.66807 25.0854 0.836171C25.1163 1.00427 25.1547 1.38721 25.1241 1.68642C24.7758 5.10162 23.2687 13.3894 22.5019 17.2145C22.1775 18.833 21.5387 19.3757 20.9202 19.4288C19.5761 19.5443 18.5555 18.5999 17.2537 17.8036C15.2167 16.5575 14.0659 15.7818 12.0886 14.5659C9.80344 13.1607 11.2848 12.3883 12.5871 11.1261C12.9279 10.7958 18.8499 5.76918 18.9645 5.3132C18.9788 5.25617 18.9921 5.04359 18.8568 4.93134C18.7215 4.81909 18.5217 4.85748 18.3776 4.888C18.1733 4.93128 14.9191 6.93846 8.61496 10.9096C7.69126 11.5015 6.85461 11.7899 6.10499 11.7747C5.2786 11.7581 3.68896 11.3387 2.50722 10.9802C1.05777 10.5406 -0.0942235 10.3081 0.00608927 9.56141C0.0583383 9.17247 0.632283 8.77472 1.72792 8.36813Z"
          fill="#858EC6"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        fill="none"
        focusable="false"
        height="24"
        viewBox="0 0 26 24"
        width="26"
      >
        <path
          clipRule="evenodd"
          d="M0.473696 23.9663H5.92744C6.18818 23.9663 6.40009 23.7826 6.40009 23.5545V7.60175C6.40009 7.37455 6.18818 7.18989 5.92744 7.18989H0.473696C0.212956 7.18989 0 7.37455 0 7.60175V23.5545C0 23.7826 0.212956 23.9663 0.473696 23.9663ZM25.6015 14.5668V23.5639C25.6015 23.7868 25.4081 23.9663 25.168 23.9663H20.1639C19.9248 23.9663 19.7304 23.7868 19.7304 23.5639V15.1965C19.7304 12.9943 18.8833 11.4908 16.7567 11.4908C15.135 11.4908 14.1708 12.5034 13.7458 13.4834C13.5915 13.8336 13.5515 14.32 13.5515 14.8118V23.5639C13.5515 23.7859 13.3571 23.9663 13.118 23.9663H8.1148C7.8747 23.9663 7.68033 23.7868 7.68128 23.5639C7.69271 21.3352 7.74321 10.5232 7.69176 7.98233C7.68795 7.7568 7.88327 7.57461 8.12528 7.57461H13.118C13.3581 7.57461 13.5515 7.75503 13.5515 7.97702V9.89796C13.5391 9.91565 13.5238 9.93423 13.5124 9.95103H13.5515V9.89796C14.3318 8.78272 15.7229 7.18989 18.8433 7.18989C22.706 7.18989 25.6015 9.5327 25.6015 14.5668ZM6.4002 2.99624C6.4002 4.65012 4.96728 5.99158 3.19966 5.99158C1.43302 5.99158 0.000103205 4.65012 0.000103205 2.99624C0.000103205 1.34145 1.43302 0 3.19966 0C4.96728 0 6.4002 1.34145 6.4002 2.99624Z"
          fill="#858EC6"
          fillRule="evenodd"
          opacity="0.8"
        />
      </svg>
    </div>
  </div>
</div>
`;
